### DISCUSSION TITLE
Exploring the Future of the PyNE ENSDF Processing Module and the endf-python Repository

### DISCUSSION THREAD

**Discussion by ahnaf-tahmid-chowdhury** [paulromano/endf-python#3] [General]
While [reviewing](https://github.com/pyne/pyne/issues/1476#issuecomment-1646636737) the PyNE documentation updates, I came across the `11-ensdf-processing.ipynb` file. Unfortunately, the web links provided in the notebook appear to be outdated, possibly due to frequent updates on the website. Given that I haven't been involved with this project from its inception, I lack insight into the original appearance of the website. However, I've recently come across your `endf-python` repository.

I'm interested in whether you intend to prioritize the development of the `endf-python` repository and if you could offer some guidance on how to handle the [`pyne.ensdf_processing`](https://github.com/pyne/pyne/blob/develop/pyne/ensdf_processing.py) module. Perhaps we can consider transitioning to the `endf-python` repository for these purposes. This would involve discussions and decisions about whether it's feasible to migrate the existing functionality, or whether it makes more sense to integrate the capabilities of the `ensdf_processing` module into the `endf-python` repository.

Ultimately, the decision to close the `ensdf_processing` module would depend on factors such as the community's needs, the functionality provided by the `endf-python` repository, and the resources available for migration and maintenance.

I'm interested in providing suggestions and guidance on how to best manage this situation, taking into account the potential benefits and challenges of migrating or integrating the existing functionality. Let's discuss potential strategies to ensure the continued success of the project and the best outcome for the community.

**Comment by paulromano** [comment]
Hi @ahnaf-tahmid-chowdhury and thanks for your interest in this project. I do intend to keep this project up to date and continue development on it as time allows. However, this project is specifically for the ENDF format. ENSDF files have their own format that is much different from ENDF, and so I would not consider it to be within scope here. I think the best possibility would be for someone to maintain a Python package specifically for ENSDF files. Perhaps @brown170 might have some insights or thoughts about this.

**Reply by brown170** [reply]
Hi, I don't know much about the PyNE ENSDF module other than it is based on the now-legacy ENSDF format.  The ENSDF project is coming to the end of a 3 year modernization project that included re-envisioning the ENSDF format in JSON and revamping all of the tools that interact with ENSDF.  There is also an ENSDF API in Python.  I do not know the timeline for open sourcing the entire system as the project is only wrapping in in September and then comes the release paperwork.

**Comment by ahnaf-tahmid-chowdhury** [comment]
> Hi, I don't know much about the PyNE ENSDF module other than it is based on the now-legacy ENSDF format. The ENSDF project is coming to the end of a 3 year modernization project that included re-envisioning the ENSDF format in JSON and revamping all of the tools that interact with ENSDF. There is also an ENSDF API in Python. I do not know the timeline for open sourcing the entire system as the project is only wrapping in in September and then comes the release paperwork.

Thank you @brown170 for the updated news, it's great to hear about the modernization project for ENSDF! The migration to a JSON format and the development of revamped tools and an API for ENSDF are significant steps toward enhancing the accessibility and usability of ENSDF data. Looking forward to the release in September!
